Docs for Beta (#354)
configcat/docs: website/docs/advanced/config-v2.md

---
id: config-v2
title: Config V2 Overview
description: Config V2 is the next generation of ConfigCat. It comes with a new dashboard, API, SDKs, and features.
---

Config V2 is a new version of ConfigCat. It comes with a new dashboard, API, SDKs, and features.

## What's new?

- A bunch of new features and improvements listed below.
- New config editor UI on the Dashboard.
- [New config JSON schema.](https://github.com/configcat/config-json)
- New API: [See the API Docs.](https://api.configcat.com/docs/)
- New SDKs: [See the supported SDK versions.](advanced/config-v2-sdk-compatibility)

## How to migrate from Config V1 to Config V2?

See the [Config V2 Migration Guide](advanced/config-v2-migration-guide). If you get stuck or have any questions about the migration, [contact us](https://configcat.com/support/).

## New features

### AND conditions

> With AND conditions, you can create more complex targeting rule sets for your feature flags. You can add multiple conditions to a targeting rule and they will be evaluated as a logical AND.

Reviewer: nit: Talán még könnyebben érthető lesz, ha beleszövünk egy példát:

```suggestion
With AND conditions, you can define more complex targeting rules, such as "serve this value for the users who use my Android app and whose email domain is '@example.com'".

You can add multiple conditions to a targeting rule and they will be evaluated as a logical AND.
```

### New comparators

> With the new comparators, you can create date based targeting rules and schedule releases, compare arrays etc.

Reviewer: A "schedule releases"-t inkább kihagynám, mert erről az időzített ff value change feature-re asszociálhatnak, ami egyelőre nincs.

```suggestion
With the new comparators, you can create targeting rules which are based on dates, based on comparing arrays etc.
```

- New text and sensitive text comparators: EQUALS, NOT EQUALS, STARTS WITH ANY OF, ENDS WITH ANY OF, NOT STARTS WITH ANY OF, NOT ENDS WITH ANY OF.
- New array comparators: ARRAY CONTAINS ANY OF, ARRAY NOT CONTAINS ANY OF.
- New date comparators: BEFORE, AFTER.

### Prerequisite flags

> With prerequisite flags, you can create feature flags that depend on other feature flags. Prerequisite feature flags are particularly useful for managing complex feature dependencies and ensuring a smooth user experience during feature rollouts.

Reviewer: A "master switch"/"master flag" kulcsszót érdemes lenne beleszőni, szerintem abból a legtöbben rögtön megértik, hogy mi ez. (Bár az ilyen szavak használata nyilván nem túl píszí mostanában 😅 )

### Comparison value hints

> With comparison value hints, you can add hints to your comparison values. This way you can add a description to your comparison value list items that helps you remember what they are for.

Reviewer: nit: Kicsit redundánsnak tűnik az eleje.

```suggestion
With comparison value hints, you can associate arbitrary text with your comparison values. This way you can add a description to your comparison value list items that helps you remember what they are for.
```

### Percentage options for targeting rules

> You can add percentage options to your targeting rules. This is useful if you want to create percentage options based on your user attributes.

Reviewer: Ebbe is tennék példát:

```suggestion
You can add percentage options to your targeting rules. This is useful if you want to create more complex targeting rules, such as "serve this value for the 20% of the users who use my Android app and serve that value for the 80% of them".

In other words, when the targeting rule matches, don't serve a simple value but split the selected group of users into additional buckets. 
```

### Custom percentage attributes

> With custom percentage attributes, you can create percentage options based on custom attributes. This way you can create percentage options based on any of your user attributes.

Reviewer: Itt is tök jó lenne egy példa, orggal vagy tenanttal, stb.
